### ISSUE TITLE
handling colors and other modifiable attributes

### ISSUE THREAD

**Issue by kushalkolar** [kushalkolar/fastplotlib#77]
The way colors are handled by `Graphic` is kinda of wonky now, I'm thinking of making a base class `GraphicAttribute` which is indexable, this would allow intuitive fine grained color control. It would probably also fix the current wonkyness of `ScatterGraphic`.

example:
```python3
plot = Plot()

line_graphic = plot.line(data=<sine_wave>)

line_graphic.colors[50:100] = "red"
line_graphic.colors[100:] = (0, 1, 0, 1)
line_graphic.colors = "w"
...
```

This would basically call `pygfx` geometry/material `colors.data` and the `update_range()` for only the indices modified

@EricThomson @clewis7 thoughts?

@clewis7 this will get rid of all the current `update_<graphic_attr>` methods, but we can change that later. For interactivity we can just allow anything that is a `GraphicAttribute` instance to be a modifiable feature.

**Comment by EricThomson** [comment]
This seems great I really like it! Would make certain things really intuitive and simple.

**Comment by kushalkolar** [comment]
Could also add sizes, at least for scatter, might also be possible for line

**Comment by kushalkolar** [comment]
@clewis7 once #61 is merged we need to think of how to trigger events when a `GraphicAttribute` changes its data, will have to work for `Graphic` and `GraphicCollection` objects.

**Comment by kushalkolar** [comment]
for now thing left to do is sizes

**Comment by kushalkolar** [comment]
mostly implemented in #78 and #88, this issue superseeded by #85
